### DISCUSSION TITLE
Improve back-end architecture with domain-driven-hexagon

### DISCUSSION THREAD

**Discussion by ElliottCuvelier** [twentyhq/twenty#872] [Ideas]
To scale this project in the future, I recommend following the Domain Driven Design and Hexagon Architecture principles. A great example of these principles implemented with NestJS can be found in this repository: https://github.com/Sairyss/domain-driven-hexagon.

**Comment by FelixMalfait** [comment]
Thanks for sharing this @ElliottCuvelier! I hadn't seen this discussion
Definitely a cool concept to keep in mind as we refactor the backend!

cc @magrinj fyi
